### PR TITLE
Build the documentation pages

### PR DESCRIPTION
**Pull request by heshamMassoud**
#### Summary

Builds the documentation pages 🎉 : https://commercetools.github.io/commercetools-sync-java

#### Hints for Review
1. Use the `ru.vyarus.mkdocs' version '1.1.0` gradle plugin to build and publish the doc pages.
2. Install python in the travis script for the gradle plugin to be able to build the pages.
3. Change all relative src paths to absolute in READMEs.
4. Add a custom README page which is only for the doc pages.
5. Reformat [Release notes](https://commercetools.github.io/commercetools-sync-java/doc/test/RELEASE_NOTES/)

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+### Prerequisites
+ 
+ - Make sure you have `JDK 8` installed.
+ - a target CTP project to which your source of data would be synced to.
+ - [commercetools-jvm-sdk](https://github.com/commercetools/commercetools-jvm-sdk) as a dependency in your JVM-based 
+  application. (Make sure to use a version `>= 1.35.0`).
+
+
+### Installation
+There are multiple ways to download the commercetools sync dependency, based on your dependency manager. Here are the
```
Comment (ahalberkamp) [inline docs/README.md]: There are multiple ways to add the commercetools sync dependency to your project

File: docs/usage/CATEGORY_SYNC.md [hunk 2]
```diff
@@ -75,12 +75,14 @@ stats.getReportMessage();
...
+
  1. The sync processing time should not take into account the time between supplying batches to the sync. 
  2. It is not known by the sync which batch is going to be the last one supplied.
 
-More examples of how to use the sync 
-1. From another CTP project as a source can be found [here](/src/integration-test/java/com/commercetools/sync/integration/ctpprojectsource/categories/CategorySyncIT.java).
-2. From an external source can be found [here](/src/integration-test/java/com/commercetools/sync/integration/externalsource/categories/CategorySyncIT.java).
+##### More examples of how to use the sync
+
+1. From another CTP project as a source can be found [here](https://github.com/commercetools/commercetools-sync-java/tree/master/src/integration-test/java/com/commercetools/sync/integration/ctpprojectsource/categories/CategorySyncIT.java).
```
Comment (ahalberkamp) [inline docs/usage/CATEGORY_SYNC.md]: ... From

File: docs/usage/CATEGORY_SYNC.md [hunk 3]
```diff
@@ -75,12 +75,14 @@ stats.getReportMessage();
...
+2. From an external source can be found [here](https://github.com/commercetools/commercetools-sync-java/tree/master/src/integration-test/java/com/commercetools/sync/integration/externalsource/categories/CategorySyncIT.java).
```
Comment (ahalberkamp) [inline docs/usage/CATEGORY_SYNC.md]: ... From

File: docs/README.md (hunk 1 above)
Comment (heshamMassoud) [reply docs/README.md]: f34739b

File: docs/usage/CATEGORY_SYNC.md (hunk 2 above)
Comment (heshamMassoud) [reply docs/usage/CATEGORY_SYNC.md]: f34739b

File: docs/usage/CATEGORY_SYNC.md (hunk 3 above)
Comment (heshamMassoud) [reply docs/usage/CATEGORY_SYNC.md]: f34739b

**Comment by ahalberkamp** [comment]
Maybe you can add some Tutorial/how to`s to the documentation

File: README.md [hunk 4]
```diff
@@ -12,6 +12,7 @@ Java Library used to import and/or sync (taking care of changes) data into one o
 as CSV, XML, JSON, etc.. or even from an already existing commercetools project.
 
 Currently this library supports synchronising the following entities in commercetools
+    
  - [Categories](/docs/usage/CATEGORY_SYNC.md)
```
Comment (lojzatran) [inline README.md]: Why you don't change the links here to absolute?

**Comment by heshamMassoud** [comment]
> What about file BUILD.md:89? You have there /gradle-scripts/maven-publish.gradle, but maybe you want an absolute path?

`BUILD.md` Is also only accessible from github. It's not supposed to be on the doc pages.

Only links which are on md docs which can be seen from the doc pages have to have absolute links due the different base url of the doc pages and the gh repo. Otherwise, documents which are not viewable on doc pages but only gh repo (like `docs/BUILD.md` and `README.md`) can have relative paths.

File: mkdocs.yml [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+  - User guide:
+      - Quick Start: usage/QUICK_START.md
+      - Product Sync: usage/PRODUCT_SYNC.md
+      - ProductType Sync: usage/PRODUCT_TYPE_SYNC.md
+      - Category Sync: usage/CATEGORY_SYNC.md
+      - InventoryEntry Sync: usage/INVENTORY_SYNC.md
+      - Advanced:
+        - Sync Options: usage/SYNC_OPTIONS.md
+        - Usage Tips: usage/IMPORTANT_USAGE_TIPS.md
+  - Javadoc: https://commercetools.github.io/commercetools-sync-java/v/v1.0.0-M14/
```
[the remark this answers is not in the document]
Comment (heshamMassoud) [reply mkdocs.yml]: I can't find it anywhere in the documentation of mkDocs how to do it honestly.

**Pull request merged**